src/commands/txCommands/util/gitHelpers.js

```diff
@@ -21,12 +22,49 @@ const commit$ = (commitMessage, args) => {
 			child_process.execSync('git add .');
 			const command = `git commit -m ${JSON.stringify(commitMessage)} ${args}`;
 			return child_process$(command)
-			.do(() => {
-				console.log(chalk.green(command));
-			});
+				.do(() => {
+					console.log(chalk.green(command));
+				});
 		});
 };
 
+const devGitBranch$ = Rx.Observable.bindNodeCallback(child_process.exec)(
+	'git rev-parse --abbrev-ref HEAD'
+)
+	.pluck(0)
+	.map(str => str.slice(0, -1));
+
+// Branch whether in dev or CI. Replaces forward slashes because
+// branch names are used as transifex resource names and resource
+// names need to work as valid url paths
+const gitBranch$ = Rx.Observable
+	.if(
+	() => process.env.TRAVIS_PULL_REQUEST_BRANCH,
+	Rx.Observable.of(process.env.TRAVIS_PULL_REQUEST_BRANCH),
+	devGitBranch$
+	)
+	.map(branchname => branchname.replace(/\//g, '_'));
+
+const branchResourceExists$ = Rx.Observable
+	.zip(txlib.resources$, gitBranch$)
+	.map(([resources, branch]) => resources.indexOf(branch) > -1);
+
+// don't run against master! will delete trn content on transifex
+const branchCheck$ = gitBranch$.do(branchName => {
+	if (branchName === 'master') {
+		console.log(
+			'do not run this script on master. it will kill the master resource on Transifex.'
+		);
+		process.exit(0);
```

Comment:
non-zero exit code is probably more useful - don't want any parent scripts/processes to think the script succeeded